### ISSUE TITLE
Configuration not being validated; obscure error when config missing

### ISSUE THREAD

**Issue by coreyward** [sanity-io/gatsby-source-sanity#92]
When an invalid configuration is provided, I would expect one of these errors to be thrown: https://github.com/sanity-io/gatsby-source-sanity/blob/7b79f130bc4c27fd1719e847659644b11f5b2d43/src/gatsby-node.ts#L234-L244

Instead, the following error is thrown:

> Error: ENOENT: no such file or directory, open 'node_modules/gatsby-source-sanity/src/gatsby-node.ts'

Example configuration:

```js
module.exports = {
  plugins: [
    {
      resolve: "gatsby-source-sanity",
      options: {
        projectId: undefined,
        dataset: undefined,
        token: undefined,
        watchMode: true
      }
    }
  ]
};
```

**Comment by rexxars** [comment]
Interesting. Gatsby tries to resolve the file referenced by the source map, and fails with a less than helpful message.
It seems we should be using the `reporter` instead of throwing errors anyway, so will change it to that. Thanks for filing the issue!

**Comment by rexxars** [comment]
Fixed in e930d670af238238915fe642b78260392c6f95ed
